frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx

```diff
@@ -75,48 +58,70 @@ const VMRow = ({ obj: vm }: React.ComponentProps<typeof ResourceRow>) => {
       <div className={otherRowSize}>
         <ResourceLink kind={NamespaceModel.kind} name={namespace} title={namespace} />
       </div>
-      {/* TODO(mlibra): migrate VM status in a follow-up
-    <div className={mainRowSize}>
-        <WithResources resourceMap={resourceMap} loaderComponent={() => DASHES}>
-          <VmStatus vm={vm}/>
-        </WithResources>
-    </div>
-    */}
+      <div className={mainRowSize}>
+        {loadErrorMsg || (
+          <VmStatus vm={vm} pods={_.get(pods, 'data')} migrations={_.get(migrations, 'data')} />
+        )}
+      </div>
       {/* TODO(mlibra): migrate actions in a follow-up
       <div className="dropdown-kebab-pf">
       <ResourceKebab actions={menuActions} kind={VirtualMachineModel.kind} resource={vm} resources={[
-        getResource(VirtualMachineInstanceModel, { name, namespace, isList: false }),
-        getResource(VirtualMachineInstanceMigrationModel, {namespace}),
-        getResource(PodModel, {namespace}),
+      getResource(VirtualMachineInstanceModel, { name, namespace, isList: false }),
+      getResource(VirtualMachineInstanceMigrationModel, {namespace}),
+      getResource(PodModel, {namespace}),
       ]} />
     </div>
     */}
     </ResourceRow>
   );
 };
 
-const VMList: React.FC = (props) => <List {...props} Header={VMHeader} Row={VMRow} />;
-VMList.displayName = 'VMList';
+const VmRow = (props: VmRowProps) => {
+  const { loadError, loaded } = props;
 
-const filters = undefined;
-/* TODO(mlibra): introduce extension point for list.tsx and reenable here then.
-Beyond recent kubevirt-web-ui functionality, user experience would be improved by full-featured VM status filtering.
-To properly determine VM status, aditional objects (pods or CRs) needs to be accessed.
-Recent filtering logic around list.tsx can handle just a single "listed" object (in our case, a VirtualMachine).
-We will need to find a way how to supply additional resources there.
-
-const filters = [{
-  type: 'vm-status',
-  selected: VM_SIMPLE_STATUS_ALL,
-  reducer: getSimpleVmStatus,
-  items: VM_SIMPLE_STATUS_ALL.map(status => ({ id: status, title: VM_SIMPLE_STATUS_TO_TEXT[status] }) ),
-}];
-*/
+  if (loadError) {
+    if (loaded && loadError instanceof TimeoutError) {
+      // eslint-disable-next-line no-console
+      console.info('ResourceRow retrieval timed out.');
+      return <VmRowDetail {...props} />;
+    }
 
-type VirtualMachinesPageProps = {
-  namespace: string;
+    // eslint-disable-next-line no-console
+    console.warn('Error loading ResourceRow: ', loadError.message);
+    return <VmRowDetail loadErrorMsg="Loading error" {...props} />;
+  }
+
+  return <VmRowDetail {...props} />;
 };
 
+const VMRowFirehose = ({ obj: vm }: React.ComponentProps<typeof ResourceRow>) => {
```

Comment:
Note that once this https://github.com/openshift/console/pull/1758 merges, it will be possible to pass resources to MultiListPage so you could use that to fetch migrations and pods and avoid connecting each row separately.

Reply:
Yes, that will be much easier. I don't have any numbers, but I believe this is not a blocker for this PR as the performance is not significantly hit by wrapping every row.

Reply:
I would prefer to wait for #1758 to merge (probably today) as the new approach is more performant and does use only few websockets instead of `n`.